src/ameba/rule/layout/trailing_blank_lines.cr

```diff
@@ -13,11 +13,18 @@ module Ameba::Rule::Layout
       description "Disallows trailing blank lines"
     end
 
-    MSG = "Blank lines detected at the end of the file"
+    MSG               = "Unnecessary blank line detected at the end of the file"
+    MSG_FINAL_NEWLINE = "Final newline missing"
 
     def test(source)
-      if source.lines.size > 1 && source.lines[-2, 2].join.strip.empty?
-        issue_for({source.lines.size - 1, 1}, MSG)
+      source_lines = source.lines
+      last_source_line = source_lines.last
```

Comment:
This will raise when `lines == 0`.

Reply:
@Sija If `lines == 0` source_lines.last returns `nil`
Then in `source_lines_size == 1 && last_source_line.empty?` first condition will return false and will not checks the second condition? What should raise an error?

Reply:
> @Sija If `lines == 0` source_lines.last returns `nil`

Nope, `Array#last` is raising. See my suggestion with `return if ...` to avoid that.

Reply:
Ha, agree :+1:  My bad.